### PR TITLE
Lint

### PR DESCRIPTION
**Pull request by yakov116**
Does most of #4008

- Use table icon in `table-input`
- Use g-plane/typed-query-selector in `fetch-dom` (I have no idea what I did here, I copied what you wrote in select-dom)
- Fix `link-to-changelog-file` book color
- Use `pageDetect` to exclude in `enable-file-links-in-compare-view`
- Restore valid checkmark in options
- `GitHub Enterprise` => `GHE`

Regarding the valid token:

image

Seems like this was removed by mistake when you redid the code

### CODE REVIEW

File: source/features/enable-file-links-in-compare-view.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ void features.add(__filebasename, {
 	],
 	exclude: [
 		// Only enabled on Open/Draft PRs. Editing files doesn't make sense after a PR is closed/merged.
```
Comment (fregante) [inline source/features/enable-file-links-in-compare-view.tsx:63]:
```suggestion
		// Editing files doesn't make sense after a PR is closed/merged
```

File: source/features/enable-file-links-in-compare-view.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ void features.add(__filebasename, {
 	],
 	exclude: [
```
Comment (fregante) [inline source/features/enable-file-links-in-compare-view.tsx:62]: I just realized that this feature also doesn't make sense when the repo has been deleted but the PR is open. It appears as "unknown repository" and it's worth checking other features relying on "get head branch"

File: source/features/first-published-tag-for-merged-pr.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {buildRepoURL, getRepo} from '../github-helpers';
 import onConversationHeaderUpdate from '../github-events/on-conversation-header-update';
 
 const getFirstTag = cache.function(async (commit: string): Promise<string | undefined> => {
-	const firstTag = await fetchDom<HTMLAnchorElement>(
+	const firstTag = await fetchDom(
```
Comment (fregante) [inline source/features/first-published-tag-for-merged-pr.tsx:13]: Looks like the types are working 👍

File: source/features/link-to-changelog-file.tsx [hunk 4]
```diff
@@ -68,7 +68,7 @@ async function init(): Promise<void | false> {
 			style={{padding: '6px 16px'}}
 			role="button"
 		>
-			<BookIcon className="text-blue mr-2"/>
+			<BookIcon className="text-blue color-text-link mr-2"/>
```
Comment (fregante) [inline source/features/link-to-changelog-file.tsx:71]: Good catch. Are there other `text-` classes used anywhere?

File: source/features/safer-destructive-actions.css [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /* Move "close issue" and "cancel" buttons on authoring comments to the left */
 
 /* ...in issue comment form */
-/* the legacy way (kept for GitHub Enterprise) */
+/* the legacy way (kept for GHE) */
```
Comment (fregante) [inline source/features/safer-destructive-actions.css:4]: GHE comments should include a date or issue or we should mark ALL of them with the expiring TODOs feature:

```js
// TODO[2021-10-01]: GHE #1234
```

My guess is that this specified piece of code is at least 2 years old and can be dropped though

File: source/options.tsx [hunk 6]
```diff
@@ -22,7 +22,7 @@ function reportStatus({error, text, scopes}: Status): void {
 	if (error) {
 		tokenStatus.dataset.validation = 'invalid';
 	} else {
-		delete tokenStatus.dataset.validation;
+		tokenStatus.dataset.validation = 'valid';
```
Comment (fregante) [inline source/options.tsx:25]: This is not equivalent. `valid` means valid, the lack of it means _undefined, we don't know yet._ It shows a different icon

Comment (fregante) [reply source/options.tsx:25]: > * Restore valid checkmark in options

This isn't what I suggested

File: source/features/link-to-changelog-file.tsx (hunk 4 above)
Comment (yakov116) [reply source/features/link-to-changelog-file.tsx:71]: Nope!

File: source/features/safer-destructive-actions.css (hunk 5 above)
Comment (yakov116) [reply source/features/safer-destructive-actions.css:4]: I am going to drop it when we drop refresh.

Will be much easier to drop it all at once IMO

File: source/options.tsx (hunk 6 above)
Comment (yakov116) [reply source/options.tsx:25]: Will add.

Wording suggestion?

File: source/features/safer-destructive-actions.css (hunk 5 above)
Comment (fregante) [reply source/features/safer-destructive-actions.css:4]: This really is 2 years old though, this rule can be deleted now

Screen Shot 2021-04-07 at 12 30 49

Comment (yakov116) [reply source/features/safer-destructive-actions.css]: What about 

https://github.com/sindresorhus/refined-github/blob/edf6b1a50f1cdb55c3303869537f105b2144afa2/source/features/more-dropdown.css#L1-L7

Also 2 years old

**Comment by yakov116** [comment]
New and updated wording

image


💤 good night will continue tomorrow

File: source/options.tsx (hunk 6 above)
Comment (fregante) [reply source/options.tsx]: No wording changes required. The `p` becomes `li`

File: distribution/options.html [hunk 7]
```diff
@@ -22,6 +22,7 @@
 		The token enables <a href="https://github.com/sindresorhus/refined-github/search?q=github-helpers+api" target="_blank">some features</a> to <strong>read</strong> data from public repositories
 	</p>
 	<ul>
+		<li data-validation data-scope="valid_token">The token is valid and it is 2 AM and I cannot think of better wording 💤 💤 💤
```
Comment (fregante) [inline distribution/options.html:25]:
```suggestion
		<li data-validation data-scope="valid_token">The token enables <a href="https://github.com/sindresorhus/refined-github/search?q=github-helpers+api" target="_blank">some features</a> to <strong>read</strong> data from public repositories

```

Comment (yakov116) [reply distribution/options.html]: One second I dont understand having a token with nothing enabled allows some features to run?

Comment (fregante) [reply distribution/options.html]: Yes

File: distribution/options.html [hunk 8]
```diff
@@ -22,6 +22,7 @@
 		The token enables <a href="https://github.com/sindresorhus/refined-github/search?q=github-helpers+api" target="_blank">some features</a> to <strong>read</strong> data from public repositories
```
Comment (fregante) [inline distribution/options.html:22]: This can be dropped

File: source/features/cross-deleted-pr-branches.tsx [hunk 9]
```diff
@@ -18,9 +18,9 @@ function init(): void | false {
 
 	const deletedBranchName = lastBranchAction.textContent!.trim();
 
-	const headReferenceLink = select('.head-ref a')!;
-	const repoRootUrl = headReferenceLink.href.split('/', 5).join('/');
-	const repoIsDeleted = headReferenceLink.textContent === 'unknown repository';
+	const headReference = select('.head-ref')!;
+	const repoIsDeleted = headReference.textContent === 'unknown repository';
+	const repoRootUrl = select('a', headReference)?.href.split('/', 5).join('/');
```
Comment (fregante) [inline source/features/cross-deleted-pr-branches.tsx:23]:
```suggestion
	const repoRootUrl = select('a', headReference)?.pathname.split('/', 3).join('/');
```

Also I think if `repoRootUrl === undefined` then `repoIsDeleted === true`. In short, we don't need `repoIsDeleted` nor its exact check. Right?

Comment (yakov116) [reply source/features/cross-deleted-pr-branches.tsx:23]: I am not sure I think github changed how they handle delete repos.

I will make a GHE comment and remove it when we drop refresh.

Comment (fregante) [reply source/features/cross-deleted-pr-branches.tsx:23]: No need to keep dead code around when we’re in doubt. We’re not fully sure this works on GHE either

Comment (yakov116) [reply source/features/cross-deleted-pr-branches.tsx:23]: I am going to send this in a separate PR. This is a real fix.

File: source/features/safer-destructive-actions.css [hunk 10]
```diff
@@ -1,12 +1,5 @@
 /* Move "close issue" and "cancel" buttons on authoring comments to the left */
```
Comment (yakov116) [inline source/features/safer-destructive-actions.css:2]:
```suggestion

/* ...in issue comment form */
```

**Comment by yakov116** [comment]
@fregante I want to merge this. Are you ok with that?

**Pull request merged**